### DISCUSSION TITLE
Add ability to enable Copilot based on directory path in VS Code Extension

### DISCUSSION THREAD

**Discussion by sw23** [community/community#48065] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

The VS Code extension has a way to enable/disable co-pilot based on programming language; it'd be great if there was similar support for enabling or disabling the extension based on a directory path.

Here's the documentation section for controlling based on programming language:
https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-visual-studio-code#enabling-or-disabling-github-copilot-for-specific-languages

There may be certain directories with sensitive code or with business requirements preventing their use with Copilot, so it'd be nice to be able to exclude those directories. Alternatively, in some cases it may be preferable to disable Copilot by default and to only opt-in certain directories.

**Comment by Arinerron** [comment]
This is important to me, please implement this.

I would prefer to blacklist directories than to have to whitelist individual directories, though. I want to Copilot (in neovim) but can't risk leaking work-related source code which I keep in one directory.

**Comment by dls314** [comment]
I would enjoy being able to disable co-pilot suggestions for files by a glob like `**/package.json`. I've found that suggestions for npm package versions in package.json files can suggest versions of the packages that don't exist.

**Comment by zeckem19** [comment]
bump, this would be a great feature
